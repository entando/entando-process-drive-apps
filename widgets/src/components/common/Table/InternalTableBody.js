import PropTypes from 'prop-types';
import React from 'react';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';

import columnType from 'types/columnType';
import TableBulkSelectContext from 'components/common/Table/TableBulkSelectContext';
import withStyles from '@material-ui/core/styles/withStyles';
import InternalTableCell from './InternalTableCell';

const StyledTableRowHover = withStyles(
  {
    root: {
      color: '#e7eaec',
      '& > *': {
        backgroundColor: '#ffffff',
      },
      '&.activeTask > *': {
        backgroundColor: '#e8f3f9',
      },
      '&:hover > *': {
        backgroundColor: '#f9f8f8',
        cursor: 'pointer',
      },
    },
    selected: {
      '& > *': {
        backgroundColor: '#f4f6f7',
      },
      '&.activeTask > *': {
        backgroundColor: '#dce7ed',
      },
    },
  },
  { name: 'StyledTableRowHover' }
)(TableRow);

const InternalTableBody = ({ columns, rows, emptyRows, rowHeight, onRowClick, activeTaskId }) => {
  return (
    <TableBody>
      {rows.map((row, idx) => (
        <TableBulkSelectContext.Consumer key={JSON.stringify(row)}>
          {({ selectedRows, rowAccessor }) => (
            <StyledTableRowHover
              style={{ height: rowHeight, cursor: row.onClick ? 'pointer' : 'initial' }}
              hover
              className={row.id === activeTaskId ? 'activeTask' : ''}
              selected={selectedRows.has(row[rowAccessor])}
              onClick={e => {
                onRowClick(row, idx, e);
              }}
            >
              {columns.map(column => (
                <InternalTableCell key={JSON.stringify(column)} column={column} row={row} />
              ))}
            </StyledTableRowHover>
          )}
        </TableBulkSelectContext.Consumer>
      ))}
      {emptyRows > 0 && (
        <TableRow style={{ height: rowHeight * emptyRows, background: 'white' }}>
          <TableCell colspan={columns.length} />
        </TableRow>
      )}
    </TableBody>
  );
};

InternalTableBody.propTypes = {
  columns: PropTypes.arrayOf(columnType),
  rows: PropTypes.arrayOf(PropTypes.shape({})),
  emptyRows: PropTypes.number,
  rowHeight: PropTypes.number,
  onRowClick: PropTypes.func,
  activeTaskId: PropTypes.string,
};

InternalTableBody.defaultProps = {
  columns: [],
  rows: [],
  emptyRows: 0,
  rowHeight: 55,
  onRowClick: () => {},
  activeTaskId: '',
};

export default InternalTableBody;
