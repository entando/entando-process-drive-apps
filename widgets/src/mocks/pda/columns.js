export default {
  payload: [
    'id',
    'name',
    'description',
    'createdBy',
    'createdAt',
    'dueTo',
    'status',
    'owner',
    'priority',
    'subject',
    'activatedAt',
    'skipable',
    'processDefinitionId',
    'processId',
    'parentId',
  ],
  errors: {},
};
