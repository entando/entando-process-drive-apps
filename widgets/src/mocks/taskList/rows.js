const rows = [
  {
    inlimit: 'true',
    application: 'com.myspace.mortgage_app.Application@15f103e7	',
    incdownpayment: 'Human Resources',
    id: 1,
    name: 'Final Approval',
    subject: 'Test',
    desc: '',
    status: 'Reserved',
    priority: 0,
    skipable: 'false',
    owner: 'pamAdmin',
    created: new Date('10/12/2016'),
    activated: new Date('10/12/2016'),
    processinstanceid: 150,
    processdefinitionid: 'Mortgage_Process.MortgageApprovalProcess',
    containerid: 'mortgage-process_1.0.0-SNAPSHOT',
    parentid: -1,
  },
  {
    inlimit: 'true',
    application: 'com.myspace.mortgage_app.Application@3423423	',
    incdownpayment: 'Human Resources',
    id: 2,
    name: 'Final Approval',
    subject: 'Test',
    desc: '',
    status: 'Reserved',
    priority: 0,
    skipable: 'false',
    owner: 'pamAdmin',
    created: new Date('10/12/2016'),
    activated: new Date('10/12/2016'),
    processinstanceid: 150,
    processdefinitionid: 'Mortgage_Process.MortgageApprovalProcess',
    containerid: 'mortgage-process_1.0.0-SNAPSHOT',
    parentid: -1,
  },
  {
    inlimit: 'true',
    application: 'com.myspace.mortgage_app.Application@23434253	',
    incdownpayment: 'Human Resources',
    id: 3,
    name: 'Final Approval',
    subject: 'Test',
    desc: '',
    status: 'Reserved',
    priority: 0,
    skipable: 'false',
    owner: 'pamAdmin',
    created: new Date('10/12/2016'),
    activated: new Date('10/12/2016'),
    processinstanceid: 150,
    processdefinitionid: 'Mortgage_Process.MortgageApprovalProcess',
    containerid: 'mortgage-process_1.0.0-SNAPSHOT',
    parentid: -1,
  },
  {
    inlimit: '',
    application: 'com.myspace.mortgage_app.Application@23523533	',
    incdownpayment: 'Human Resources',
    id: 4,
    name: 'Final Approval',
    subject: 'Test',
    desc: '',
    status: 'Reserved',
    priority: 0,
    skipable: 'false',
    owner: 'pamAdmin',
    created: new Date('10/12/2016'),
    activated: new Date('10/12/2016'),
    processinstanceid: 150,
    processdefinitionid: 'Mortgage_Process.MortgageApprovalProcess',
    containerid: 'mortgage-process_1.0.0-SNAPSHOT',
    parentid: -1,
  },
  {
    inlimit: '',
    application: 'com.myspace.mortgage_app.Application@23523535	',
    incdownpayment: 'Human Resources',
    id: 5,
    name: 'Final Approval',
    subject: 'Test',
    desc: '',
    status: 'Reserved',
    priority: 0,
    skipable: 'false',
    owner: 'pamAdmin',
    created: new Date('10/12/2016'),
    activated: new Date('10/12/2016'),
    processinstanceid: 150,
    processdefinitionid: 'Mortgage_Process.MortgageApprovalProcess',
    containerid: 'mortgage-process_1.0.0-SNAPSHOT',
    parentid: -1,
  },
  {
    inlimit: 'true',
    application: 'com.myspace.mortgage_app.Application@23523535	',
    incdownpayment: 'Human Resources',
    id: 6,
    name: 'Final Approval',
    subject: 'Test',
    desc: '',
    status: 'Reserved',
    priority: 0,
    skipable: 'false',
    owner: 'pamAdmin',
    created: new Date('10/12/2016'),
    activated: new Date('10/12/2016'),
    processinstanceid: 150,
    processdefinitionid: 'Mortgage_Process.MortgageApprovalProcess',
    containerid: 'mortgage-process_1.0.0-SNAPSHOT',
    parentid: -1,
  },
  {
    inlimit: 'true',
    application: 'com.myspace.mortgage_app.Application@2352362362	',
    incdownpayment: 'Human Resources',
    id: 7,
    name: 'Final Approval',
    subject: 'Test',
    desc: '',
    status: 'Reserved',
    priority: 0,
    skipable: 'false',
    owner: 'pamAdmin',
    created: new Date('10/12/2016'),
    activated: new Date('10/12/2016'),
    processinstanceid: 150,
    processdefinitionid: 'Mortgage_Process.MortgageApprovalProcess',
    containerid: 'mortgage-process_1.0.0-SNAPSHOT',
    parentid: -1,
  },
  {
    inlimit: 'true',
    application: 'com.myspace.mortgage_app.Application@2362362	',
    incdownpayment: 'Human Resources',
    id: 8,
    name: 'Final Approval',
    subject: 'Test',
    desc: '',
    status: 'Reserved',
    priority: 0,
    skipable: 'false',
    owner: 'pamAdmin',
    created: new Date('10/12/2016'),
    activated: new Date('10/12/2016'),
    processinstanceid: 150,
    processdefinitionid: 'Mortgage_Process.MortgageApprovalProcess',
    containerid: 'mortgage-process_1.0.0-SNAPSHOT',
    parentid: -1,
  },
  {
    inlimit: 'false',
    application: 'com.myspace.mortgage_app.Application@2362362	',
    incdownpayment: 'Human Resources',
    id: 9,
    name: 'Final Approval',
    subject: 'Test',
    desc: '',
    status: 'Reserved',
    priority: 0,
    skipable: 'false',
    owner: 'pamAdmin',
    created: new Date('10/12/2016'),
    activated: new Date('10/12/2016'),
    processinstanceid: 150,
    processdefinitionid: 'Mortgage_Process.MortgageApprovalProcess',
    containerid: 'mortgage-process_1.0.0-SNAPSHOT',
    parentid: -1,
  },
  {
    inlimit: 'false',
    application: 'com.myspace.mortgage_app.Application@7534542	',
    incdownpayment: 'Human Resources',
    id: 10,
    name: 'Final Approval',
    subject: 'Test',
    desc: '',
    status: 'Reserved',
    priority: 0,
    skipable: 'false',
    owner: 'pamAdmin',
    created: new Date('10/12/2016'),
    activated: new Date('10/12/2016'),
    processinstanceid: 150,
    processdefinitionid: 'Mortgage_Process.MortgageApprovalProcess',
    containerid: 'mortgage-process_1.0.0-SNAPSHOT',
    parentid: -1,
  },
];

export default rows;
