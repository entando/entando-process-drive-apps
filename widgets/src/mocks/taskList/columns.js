import { compareDates, compareNumbers, compareStrings } from 'components/common/Table/utils';
import DateCell from 'components/common/Table/custom/DateCell';
import ActionCell from 'components/common/Table/custom/ActionCell';

const columns = [
  {
    header: 'Action',
    accessor: 'action',
    customCell: ActionCell,
    styles: {
      position: 'sticky',
      left: 0,
      background: 'white',
      width: 20,
      zIndex: 100,
      borderRight: '1px solid #eee',
      paddingRight: 16,
      textAlign: 'center',
    },
  },
  {
    header: 'Inlimit',
    accessor: 'inlimit',
    sortFunction: compareStrings,
  },
  {
    header: 'Application',
    accessor: 'application',
    sortFunction: compareStrings,
  },
  {
    header: 'Inc Down Payment',
    accessor: 'incdownpayment',
    sortFunction: compareStrings,
  },
  {
    header: 'ID',
    accessor: 'id',
    sortFunction: compareNumbers,
  },
  {
    header: 'Name',
    accessor: 'name',
    sortFunction: compareStrings,
  },
  {
    header: 'Subject',
    accessor: 'subject',
    sortFunction: compareStrings,
  },
  {
    header: 'Description',
    accessor: 'desc',
    sortFunction: compareStrings,
  },
  {
    header: 'Status',
    accessor: 'status',
    sortFunction: compareStrings,
  },
  {
    header: 'Priority',
    accessor: 'priority',
    sortFunction: compareNumbers,
  },
  {
    header: 'Skipable',
    accessor: 'skipable',
    sortFunction: compareStrings,
  },
  {
    header: 'Owner',
    accessor: 'owner',
    sortFunction: compareStrings,
  },
  {
    header: 'Created',
    accessor: 'created',
    sortFunction: compareDates,
    customCell: DateCell('created'),
  },
  {
    header: 'Activated',
    accessor: 'activated',
    sortFunction: compareDates,
    customCell: DateCell('activated'),
  },
  {
    header: 'Process Instance ID',
    accessor: 'processinstanceid',
    sortFunction: compareStrings,
  },
  {
    header: 'Process Definition ID',
    accessor: 'processdefinitionid',
    sortFunction: compareStrings,
  },
  {
    header: 'Container ID',
    accessor: 'containerId',
    sortFunction: compareStrings,
  },
  {
    header: 'Parent ID',
    accessor: 'parentid',
    sortFunction: compareNumbers,
  },
];

export default columns;
