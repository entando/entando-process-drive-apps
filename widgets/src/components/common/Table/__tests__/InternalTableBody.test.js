import React from 'react';
import { render } from '@testing-library/react';

import InternalTableBody from 'components/common/Table/InternalTableBody';
import columns from 'mocks/taskList/columns';
import rows from 'mocks/taskList/rows';

describe('<InternalTableBody />', () => {
  it('renders snapshot correctly', () => {
    const rowsNumber = 2;
    const { container } = render(
      <table>
        <InternalTableBody rows={rows.slice(0, rowsNumber)} columns={columns} emptyRows={0} />
      </table>
    );

    expect(container).toMatchSnapshot();
    expect(document.querySelectorAll('tr').length).toBe(rowsNumber);
  });

  it('renders snapshot correctly with emptyRows', () => {
    const { container } = render(
      <table>
        <InternalTableBody rows={rows.slice(0, 2)} columns={columns} emptyRows={3} />
      </table>
    );

    expect(container).toMatchSnapshot();
  });
});
