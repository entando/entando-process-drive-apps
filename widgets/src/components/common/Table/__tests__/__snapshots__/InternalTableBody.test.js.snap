// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InternalTableBody /> renders snapshot correctly 1`] = `
<div>
  <table>
    <tbody
      class="MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root StyledTableRowHover-root-2 MuiTableRow-hover"
        style="height: 55px; cursor: initial;"
      >
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          294@mortgage-process_1.0.0-SNAPSHOT
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Correct Data
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          pamAdmin
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          2020-01-07T04:17:51Z
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          <div
            class="MuiChip-root makeStyles-root-22 makeStyles-root-23"
          >
            <span
              class="MuiChip-label"
            >
              RESERVED
            </span>
          </div>
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          pamAdmin
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          0
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          2020-01-07T04:17:51Z
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          false
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          160
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Mortgage_Process.MortgageApprovalProcess
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          -1
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
      </tr>
      <tr
        class="MuiTableRow-root StyledTableRowHover-root-2 MuiTableRow-hover"
        style="height: 55px; cursor: initial;"
      >
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          293@evaluation_1.0.0-SNAPSHOT
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          PM Evaluation
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          pamAdmin
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          2019-11-14T09:26:30Z
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          <div
            class="MuiChip-root makeStyles-root-22 makeStyles-root-54"
          >
            <span
              class="MuiChip-label"
            >
              CREATED
            </span>
          </div>
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          You need to evaluate Alessio.
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          0
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          You need to evaluate Alessio.
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          2019-11-14T09:26:30Z
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          false
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          159
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          evaluation
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          -1
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-9 MuiTableCell-body MuiTableCell-sizeSmall"
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<InternalTableBody /> renders snapshot correctly with emptyRows 1`] = `
<div>
  <table>
    <tbody
      class="MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root StyledTableRowHover-root-56 MuiTableRow-hover"
        style="height: 55px; cursor: initial;"
      >
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          294@mortgage-process_1.0.0-SNAPSHOT
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Correct Data
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          pamAdmin
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          2020-01-07T04:17:51Z
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          <div
            class="MuiChip-root makeStyles-root-76 makeStyles-root-77"
          >
            <span
              class="MuiChip-label"
            >
              RESERVED
            </span>
          </div>
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          pamAdmin
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          0
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          2020-01-07T04:17:51Z
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          false
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          160
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          Mortgage_Process.MortgageApprovalProcess
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          -1
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
      </tr>
      <tr
        class="MuiTableRow-root StyledTableRowHover-root-56 MuiTableRow-hover"
        style="height: 55px; cursor: initial;"
      >
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          293@evaluation_1.0.0-SNAPSHOT
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          PM Evaluation
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          pamAdmin
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          2019-11-14T09:26:30Z
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          <div
            class="MuiChip-root makeStyles-root-76 makeStyles-root-108"
          >
            <span
              class="MuiChip-label"
            >
              CREATED
            </span>
          </div>
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          You need to evaluate Alessio.
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          0
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          You need to evaluate Alessio.
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          2019-11-14T09:26:30Z
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          false
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          159
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          evaluation
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        >
          -1
        </td>
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
        <td
          class="MuiTableCell-root StyledTableCell-root-63 MuiTableCell-body MuiTableCell-sizeSmall"
        />
      </tr>
      <tr
        class="MuiTableRow-root"
        style="height: 165px; background: white;"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body"
          colspan="25"
        />
      </tr>
    </tbody>
  </table>
</div>
`;
